go/vt/vtgate/vcursor_impl.go

```diff
@@ -329,7 +329,7 @@ func (vc *vcursorImpl) FirstSortedKeyspace() (*vindexes.Keyspace, error) {
 
 // SysVarSetEnabled implements the ContextVSchema interface
 func (vc *vcursorImpl) SysVarSetEnabled() bool {
-	return *sysVarSetEnabled
+	return vc.GetSessionEnableSystemSettings() && *sysVarSetEnabled
```

Comment:
This is not correct, right? the default is the flag, so we don't need to && it here

Reply:
The default value of the session's `@@enable_system_settings` is set in here: https://github.com/vitessio/vitess/blob/f962627dbd19595f152ee86bb8c551e5eefd2d7c/go/vt/vtgate/plugin_mysql_server.go#L348

It uses the value that was given to VTGate with the flag `-enable_system_settings`. Then, users are free to set `@@enable_system_settings` however they want. You're right, I think we can just keep the following:

```
return vc.GetSessionEnableSystemSettings()
```